### PR TITLE
refactor(datepicker): simplify css handling of the navigation system

### PR DESCRIPTION
**Pull request by ExFlo**
Before submitting a pull request, please make sure you have at least performed the following:

 - [] read and followed the [CONTRIBUTING.md](https://github.com/ng-bootstrap/ng-bootstrap/blob/master/CONTRIBUTING.md) guide.
 - [] built and tested the changes locally.
 - [] added/updated any applicable tests.
 - [] added/updated any applicable API documentation.
 - [] added/updated any applicable demos.

### CODE REVIEW

File: src/datepicker/datepicker-navigation.ts [hunk 1]
```diff
@@ -41,25 +38,60 @@ import {NgbCalendar} from './ngb-calendar';
...
       [date]="date"
       [minDate]="minDate"
       [maxDate]="maxDate"
       [disabled] = "disabled"
       (select)="selectDate($event)">
     </ngb-datepicker-navigation-select>
-    
+
+    <ng-template *ngIf="!showSelect" ngFor let-month [ngForOf]="months" let-i="index">
+      <div class="nbg-dp-arrow first" *ngIf="i > 0"></div>
```
Comment (maxokorokov) [inline src/datepicker/datepicker-navigation.ts]: not sure what `first` and `end` classes are for

File: src/datepicker/datepicker-navigation.ts [hunk 2]
```diff
@@ -1,55 +1,99 @@
...
+    .ngb-dp-arrow-btn.right {
+      float: right;
+    }
+    .right .ngb-dp-navigation-chevron:before {
       -webkit-transform: rotate(45deg);
       -ms-transform: rotate(45deg);
       transform: rotate(45deg);
-      margin: 0 0.5rem 0 0;
+    }
+    .btn-link {
```
Comment (maxokorokov) [inline src/datepicker/datepicker-navigation.ts]: I don't think we need to override `.btn-link` at all, let's stick with default bootstrap styling
Please see #1948, where it was removed previously

File: src/datepicker/datepicker-navigation.ts [hunk 3]
```diff
@@ -1,55 +1,99 @@
...
+    .ngb-dp-arrow {
+      flex-basis: auto;
+      flex-grow: 1;
+      padding-right: 0px;
+      padding-left: 0px;
+      margin: 0px;
+      width: 2rem;
+      height: 2rem;
+    }
+    .ngb-dp-arrow-btn {
```
Comment (maxokorokov) [inline src/datepicker/datepicker-navigation.ts]: don't think you need `backgound-color` and `border-color` either, as `btn-link` should take care of it

**Comment by maxokorokov** [comment]
Argh, also the new arrows in the new calendar demos with `RTL` seem to be broken...

screen shot 2017-12-05 at 17 58 46

File: src/datepicker/datepicker-navigation.ts (hunk 3 above)
Comment (ExFlo) [reply src/datepicker/datepicker-navigation.ts]: Hi
border-color is still needed and you are right about background-color not needed.
BR

File: src/datepicker/datepicker-navigation.ts (hunk 2 above)
Comment (ExFlo) [reply src/datepicker/datepicker-navigation.ts]: I removed the override of btn-link

**Comment by ExFlo** [comment]
Hi
I replace also d-inline-block to be consistent with the changes of this PR.

File: src/datepicker/datepicker-navigation.ts (hunk 3 above)
Comment (maxokorokov) [reply src/datepicker/datepicker-navigation.ts]: I think `.btn` already has transparent border

Comment (ExFlo) [reply src/datepicker/datepicker-navigation.ts]: Let's add btn ok

Comment (maxokorokov) [reply src/datepicker/datepicker-navigation.ts]: Oh, you removed it, again the `btn btn-sm` were added by #1948 recently, so better to put it back for now. I think `btn btn-link` are supposed to be together

**Pull request closed without merging**